### DISCUSSION TITLE
vim.api.nvim_get_hl raises error on nvim 8.0

### DISCUSSION THREAD

**Discussion by Confidenceman02** [nvim-tree/nvim-tree.lua#2641] [Q&A]
NVIM v0.8.0-1210-gd367ed9b2

I'm seeing an error after updating nvim-tree to the latest version via lazy.nvim. 

```vim
Failed to run `config` for nvim-tree.lua
...are/nvim/lazy/nvim-tree.lua/lua/nvim-tree/appearance.lua:199: attempt to call field 'nvim_get_hl' (a nil value)
# stacktrace:
  - ~/.config/nvim/lua/confidenceman02/plugins/nvim-tree.lua:14 _in_ **config**
  - ~/.config/nvim/lua/confidenceman02/lazy.lua:14
```

I believe what is going on is that neovim made a change that uses `nvim_get_hl`  over `nvim__get_hl_defs`, thus the error.

The offending code as the stack trace points to is in `appearance.lua`

```lua
    local hl_from = vim.api.nvim_get_hl(0, { name = from })
    local hl_to = vim.api.nvim_get_hl(0, { name = to })
```

This was added in this [commit](https://github.com/nvim-tree/nvim-tree.lua/pull/2455/files#diff-2d73f6c60c8fe2293066a4b6ac87be3ba41805465212f3819e80eb4e566a53d4)

Because `nvim-tree` supports nvim >=8 I would expect a guard around the api change.

Messing around locally this seems to fix it.

```lua
    local hl_from = vim.api.nvim_get_hl and vim.api.nvim_get_hl(0, { name = from }) or vim.api.nvim__get_hl_defs(0)[from] or {}
    local hl_to = vim.api.nvim_get_hl and vim.api.nvim_get_hl(0, { name = to }) or vim.api.nvim__get_hl_defs(0)[to] or {}
``` 

It's a touch ugly buy you get the point hopefully.

Happy to create a pr if it's a legit bug. 

Many thanks.

**Comment by alex-courtis** [comment]
Many thanks for raising this one; I did not think about 0.8 compatibility...

Two options:
1. Tag the previous commit as `compat-nvim-0.8`
2. Use the above workaround with a `vim.fn.has "nvim-0.9" == 1`

I'm happy to go with 2 as the shim is simple enough. Can you please raise a PR? You'll need to cover the one in `builder.lua` as well - that's for the case of multiple HL on one node.

**Reply by Confidenceman02** [reply]
Thanks for the speedy reply. 

2 sounds good!

I can and will raise a PR.

Thanks again.
